src/app/components/Header.tsx

```diff
@@ -18,15 +18,54 @@
  */
 'use client';
 
-import { AppBar, css, DnaLoader, UserBadge } from '@icgc-argo/uikit';
-import Link from 'next/link';
+import { useState } from 'react';
+import {
+	AppBar,
+	AppBarMenuItem,
+	DnaLoader,
+	DropdownMenu,
+	FocusWrapper,
+	Link,
+	NavBarElement,
+	NavElement,
+	UserBadge,
+} from '@icgc-argo/uikit';
 import Image from 'next/image';
+import { usePathname } from 'next/navigation';
 import argoLogo from '/public/argo-logo.svg';
+import { css, useTheme } from '@/lib/emotion';
 import { useAuthContext } from '@/global/utils/auth';
 import LoginButton from './LoginButton';
 
 const Header = () => {
-	const { egoJwt, loggingIn } = useAuthContext();
+	const [isDropdownOpen, setDropdownOpen] = useState(false);
+	const { egoJwt, authLoading, logOut } = useAuthContext();
+	const path = usePathname();
+	const theme = useTheme();
+	const onProfilePage = path === '/landing-page';
+	const profileActive = onProfilePage && !!egoJwt.length && !authLoading;
+
+	const profileNavDetails: Array<NavElement> = [
+		{
+			active: profileActive,
+			href: '/landing-page',
+			name: 'Profile & Token',
+			LinkComp: Link,
+		},
+		{
+			isLink: false,
+			onClick: async () => {
+				setDropdownOpen(false);
+				logOut();
+			},
+			name: 'Logout',
+			active: false,
+			href: '',
+			LinkComp: Link,
+		},
+	];
```

Comment:
flawed api to `NavBarElement`, not much improvement without editing uikit
one important thing to note is that `Link` for the `LinkComp` will not work like Next.js native component (no prefetching, SPA routing might be impacted)

having both `href` and `isLink` is redundant - observation on uikit component

Reply:
Yes and there's a Type mismatch between modern `next/link` and uikit `Link` so we have to use uikit Link here

Reply:
ah I see! always the freedom to extend uikit - i'll leave that to your discretion :)